File: An2/MAP/MAP-Java/MAP-Lab6(NB)/src/Controller/Validator.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package Controller;

import Utils.MyException;
import Repository.*;
import Model.*;

/**
 *
 * @author mihai
 */
public class Validator {
    private Repository repo;
    
    public Validator(Repository repo){
        /*
         * Constructor
         * uses repository
         */
        this.repo = repo;
    }
    
    private boolean studentExists(int id) throws MyException{

        Stack<Student> students = repo.getAllElements();
        while (!students.isEmpty()){
            Student currStudent;
            currStudent = students.pop();
            if (currStudent.getID() == id)
                return true;
        }
        return false;
    }
    
    private boolean nameIsEmpty(String name){
        if (name.length() == 0) return true;
        return false;
    }
    
    public void validateStudent(Student s) throws MyException{
        if (this.nameIsEmpty(s.getName()))
            throw new MyException("Name cannot be empty!");
        if (this.studentExists(s.getID()))
            throw new MyException("Student id already exists!");
    }
}

File: An2/Sem1/MAP/MAP-C#/MAP-Lab7/MAP-Lab7/Controller/Validator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using MAP_Lab7.Repository;
using MAP_Lab7.Model;
using MAP_Lab7.Utils;

namespace MAP_Lab7.Controller
{
    class Validator
    {
        private Repository<Student> repo;

        public Validator(Repository<Student> repo)
        {
            this.repo = repo;
        }

        private bool studentExists(int id)
        {
            MAP_Lab7.Repository.Stack<Student> students = repo.getAllElements();
            while (!students.isEmpty())
            {
                Student currStudent;
                currStudent = students.pop();
                if (currStudent.getID() == id)
                    return true;
            }
            return false;
        }

        private bool nameIsEmpty(String name)
        {
            if (name.Length < 1) return true;
            else return false;
        }

        public void validateStudent(Student s)
        {
            if (this.nameIsEmpty(s.getName()))
                throw new MyException("Name cannot be empty!\n");
            if (this.studentExists(s.getID()))
                throw new MyException("Student id already exists!\n");
        }
    }
}
